backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/migration/MigrationTaskDTO.java

```diff
@@ -0,0 +1,40 @@
+package org.apache.james.backends.cassandra.migration;
+
+import java.util.function.Function;
+
+import org.apache.james.backends.cassandra.versions.SchemaVersion;
+import org.apache.james.json.DTOModule;
+import org.apache.james.server.task.json.dto.TaskDTO;
+import org.apache.james.server.task.json.dto.TaskDTOModule;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+
+class MigrationTaskDTO implements TaskDTO {
+
+    @VisibleForTesting static final Function<MigrationTask.Factory, TaskDTOModule<MigrationTask, MigrationTaskDTO>> MODULE =
+            factory -> DTOModule.forDomainObject(MigrationTask.class)
+                .convertToDTO(MigrationTaskDTO.class)
+                .toDomainObjectConverter(dto -> factory.create(new SchemaVersion(dto.targetVersion)))
+                .toDTOConverter((task, type) -> new MigrationTaskDTO(type, task.getTargetVersion().getValue()))
+                .typeName("cassandra-migration-task")
```

Comment:
And you will be able able to use the constant for the type of the task

Reply:
What do you mean?

Reply:
The migration task should have a constant TYPE used in its 'getType' method.
You could use it to define the typename.

Reply:
there are several reasons not to do that:
1/ getType returns a value for webadmin and is probably exposed to the user. It existed before task serialization so I don't have any reason the change the returned value.
2/ using the TYPE constant would require to change in into kebab-case, it doesn't go with 1/
3/ a task should not know how it's serialized (and even with the inner class strategy, as it's static, technically, it doesn't know how it's serialized for now), so binding TYPE with typeName looks weird

It's not important, but I don't feel like it improve anything.

Reply:
ok